docs/dev.md

```diff
@@ -0,0 +1,143 @@
+# Developer Information
+
+## Git
+
+### Pull Requests
+
+A PR should be, mostly, on a single topic. A common exception to this might be
+for a small PR to include small bug fixes. If a PR becomes very large, it is
+helpful to split it into several smaller PRs, if this is possible. It is fine in
+this case to base the final PR on the earlier ones, with a note such as "Based
+on #1, #2, and #6" in the PR description.
+
+### Commits
+
+Each commit should make one logical change. This makes it easier to review the
+changes, and for bisecting for regressions. Each commit must build successfully,
+and should pass all tests, this may mean marking some tests as expected failures.
+
+A git commit should have a descriptive subject line, between 50 and 72
+characters in length. If it is not possible to be under 72 be as close as possible.
+It is helpful to preface the subject with a short description of what is
+affected. For example:
+
+```
+meson: bump gtest requirement from 1.12 to 1.15
+```
+
+After the subject should come a more detailed description of the change. This
+should include the reason for the change. It may also include implementation
+details, design decisions, and issues that may not be resolved or may need to be
+resolved in the future. Lines in this section should be wrapped to 72
+characters, or if this is not possible (such as a string that is itself longer
+than 72 characters) then be as close as possible.
+
+For example:
+
+```
+meson: bumpt gtest requirement from 1.12 to 1.15
+
+gtest 1.13 fixes a bug on Windows that is affecting our CI, see
+google/googletest#1234.
+
+gtest 1.14 adds a new feature that allows for faster test execution that will be
+used in the next patch
+```
+
+Finally, if a commit fixes any github issues, they should add that,
+
+```
+Fixes: #7
+```
+
+### Merging and rebasing
+
+Rebasing should generally be preferred to merges. Rebases create a linear
+history that is easy to bisect over, and easy for a human to understand. In a
+project like cps-config that lacks large subsystems this is easy to maintain.
+
+Merges may be appropriate for large additions or refactors that have moved
+quickly and may not have a clean history. This should generally be unnecessary.
+Merges also do not go through CI, and as such it's possible to create a state
+where main before the merge builds successfully, and the topic branch before the
+merge builds successfully, but the merge commit itself is broken.
+
+Back merges (merging the main branch back into a topic branch) should never
+happen. This makes bisects particularly hard and the history very messy. A
+rebase should be done in this case
+
+## Editors
+
+Use [editorconfig](https://editorconfig.org). Many editors support editorconfig
+natively, others have plugins. Otherwise, configure your editor to use the same
+settings.
+
+## All files
+
+All files should have an [SPDX license
```

Comment:
It would be helpful to qualify what files would need a SPDX license header and what others wouldn't, unless the intention here is to say literally all files. In that case we should just fix the existing ones that don't have this header, e.g. `meson.build`, all the cps files used for unit tests, etc.